addons/GearSwap/export.lua

```diff
@@ -80,33 +84,13 @@ function export_set(options)
     end
     
     local inv = items.inventory
-    if targinv then
-        -- Load the entire inventory
-        for _,v in pairs(inv) do
-            if type(v) == 'table' and v.id ~= 0 then
-                if res.items[v.id] then
-                    item_list[#item_list+1] = {}
-                    item_list[#item_list].name = res.items[v.id][language]
-                    local potslots,slot = copy_entry(res.items[v.id].slots)
-                    if potslots then
-                        slot = res.slots[potslots:it()()].english:gsub(' ','_'):lower() -- Multi-lingual support requires that we add more languages to slots.lua
-                    end
-                    item_list[#item_list].slot = slot or 'item'
-                    if not xml then
-                        local augments = extdata.decode(v).augments or {}
-                        local aug_str = ''
-                        for aug_ind,augment in pairs(augments) do
-                            if augment ~= 'none' then aug_str = aug_str.."'"..augment:gsub("'","\\'").."'," end
-                        end
-                        if string.len(aug_str) > 0 then
-                            item_list[#item_list].augments = aug_str
-                        end
-                    end
-                else
-                    msg.addon_msg(123,'You possess an item that is not in the resources yet.')
-                end
-            end
+    local bag_list = {'inventory', 'safe', 'storage', 'locker', 'satchel', 'sack', 'case', 'wardrobe', 'safe2', 'wardrobe2'}
+    if all_items then
+        for _, bag in ipairs(bag_list) do
```

Comment:
Instead of using a custom `bag_list` which you define you should probably iterate the `items` table itself. You may need to make sure there's nothing else in there for it to work. The downside to a custom list like this is that it will need to be updated any time it changes.

Alternatively, if iterating the `items` table is not an option, you should define the bag list like this:
```lua
local bag_list = res.bags:map(function(bag) return bag.english:gsub(' ', '') end)
```

This uses the bags as they are defined in the resources, and thus will get updated automatically with every update. But I'd still prefer the above approach, it's faster and less code.